src/main/java/com/google/sps/data/TimeRange.java

```diff
@@ -0,0 +1,186 @@
+ * providing methods to make ranges easier to work with (e.g. {@code overlaps}).
+ */
+public final class TimeRange {
+  public static final int START_OF_DAY = getTimeInMinutes(0, 0);
+  public static final int END_OF_DAY = getTimeInMinutes(23, 59);
+
+  public static final TimeRange WHOLE_DAY = new TimeRange(0, 24 * 60);
+
+  /**
+   * A comparator for sorting ranges by their start time in ascending order.
+   */
+  public static final Comparator<TimeRange> ORDER_BY_START = new Comparator<TimeRange>() {
+    @Override
+    public int compare(TimeRange a, TimeRange b) {
+      return Long.compare(a.start, b.start);
+    }
+  };
+
+  /**
+   * A comparator for sorting ranges by their end time in ascending order.
+   */
+  public static final Comparator<TimeRange> ORDER_BY_END = new Comparator<TimeRange>() {
+    @Override
+    public int compare(TimeRange a, TimeRange b) {
+      return Long.compare(a.end(), b.end());
+    }
+  };
+
+  private final int start;
+  private final int duration;
+
+  private TimeRange(int start, int duration) {
+    this.start = start;
+    this.duration = duration;
+  }
+
+  /**
+   * Returns the start of the range in minutes.
+   */
+  public int start() {
+    return start;
+  }
+
+  /**
+   * Returns the number of minutes between the start and end.
+   */
+  public int duration() {
+    return duration;
+  }
+
+  /**
+   * Returns the end of the range. This ending value is the closing exclusive bound.
+   */
+  public int end() {
+    return start + duration;
+  }
+
+  /**
+   * Checks if two ranges overlap. This means that at least some part of one range falls within the
+   * bounds of another range.
+   */
+  public boolean overlaps(TimeRange other) {
+    // For two ranges to overlap, one range must contain the start of another range.
+    //
+    // Case 1: |---| |---|
+    //
+    // Case 2: |---|
+    //            |---|
+    //
+    // Case 3: |---------|
+    //            |---|
+    return this.contains(other.start) || other.contains(this.start);
+  }
+
+  /**
+   * Checks if this range completely contains another range. This means that {@code other} is a
+   * subset of this range. This is an inclusive bounds, meaning that if two ranges are the same,
+   * they contain each other.
+   */
+  public boolean contains(TimeRange other) {
+    // If this range has no duration, it cannot contain anything.
+    if (duration <= 0) {
+      return false;
+    }
+
+    // If the other range has no duration, then we must treat it like a point in time rather than a
+    // range.
+    if (other.duration <= 0) {
+      return contains(this, other.start);
+    }
+
+    // We need the inclusive end for this check in order for this case to equal true:
+    // |------|
+    //     |--|
+    int otherInclusiveEnd = other.start + other.duration - 1;
+    return contains(this, other.start) && contains(this, otherInclusiveEnd);
+  }
+
+  public boolean contains(int point) {
```

Comment:
You're taking a primitive, and you've included no documentation about it.

(this is why we try not to use primitives for time types. The signature tells us nothing)